src/verify.c

```diff
@@ -805,6 +834,26 @@ int verify_main(int argc, char **argv)
 	}
 	grabtime_stop(&times);
 
+	/* When installing, check disk space availability */
+	/* Print a warning and set a flag if we lack free space */
+	if (cmdline_option_install) {
+		long fs_free = get_available_space(path_prefix);
+		long total_size = get_manifest_list_contentsize(official_manifest->submanifests);
+
+		/* Add 10% to total_size as a 'fudge factor' */
+		if (((total_size * 1.1) > fs_free) || fs_free < 0) {
+			if (fs_free >= 0) {
+				fprintf(stderr, "Warning: Files to install larger than free space by %ldM.\n", (total_size - fs_free) / 1000 / 1000);
```

Comment:
As we aren't going to abort the installation process when we notice that we may have free space issues, can't we report the problem only on add_missing_files() if a file copy fails?

Reply:
Agreed, I can't remember why I had the warning flag - I suppose I was concerned about checking space on other types of failures, but there's no real harm checking as failures here should be fairly rare.

Fixed.